src/Microsoft.Diagnostics.Monitoring.RestServer/Controllers/DiagController.cs

```diff
@@ -150,24 +155,35 @@ public Task<ActionResult> GetGcDump(int? pid)
         }
 
         [HttpGet("logs/{pid?}")]
+        [Produces("text/event-stream", "application/x-ndjson")]
         public ActionResult Logs(int? pid, [FromQuery][Range(-1, int.MaxValue)] int durationSeconds = 30)
         {
             TimeSpan duration = ConvertSecondsToTimeSpan(durationSeconds);
             return this.InvokeService(() =>
             {
                 int pidValue = _diagnosticServices.ResolveProcess(pid);
+                
+                LogFormat format = ComputeLogFormat(Request.GetTypedHeaders().Accept);
+                if (format == default)
+                {
+                    return new StatusCodeResult((int)HttpStatusCode.NotAcceptable);
```

Comment:
Is there no base controller method for this?

Reply:
This is by design, because under normal situations the content negotiation is done by contentformatters. See https://github.com/dotnet/aspnetcore/issues/14171